java/core/src/java/org/apache/orc/TypeDescription.java

```diff
@@ -476,7 +476,11 @@ public TypeDescription withScale(int scale) {
    * @return this for method chaining
    */
   public TypeDescription setAttribute(String key, String value) {
-    attributes.put(key, value);
+    if (value == null) {
```

Comment:
This behavior should be documented in the JavaDoc as well, otherwise it is not very clear. Additionally, should there be a `removeAttribute(...)` method instead of this? 
Seems like the `setAttribute` semantics would be to set a value to `null` if a `null` is assigned to a key rather than removing the attribute with that key.

Reply:
I agree that I need to update the Javadoc. 

What is the semantics of having a null value as a type attribute? All of the examples of type attributes that I've seen, we always want a value. If we allow null values, that means that we need to put null checks everywhere we pull a value out. If we guarantee that they aren't null, that simplifies things.

So I guess we could have it throw on getting a null value and add the removeAttribute if you think that is cleaner. Thoughts?

Reply:
I think `removeAttribute` is more clear and I agree with just with the non-null guarantee. Thanks for the changes!